daemon/algod/api/server/v2/handlers_test.go

```diff
@@ -141,8 +143,8 @@ func makeTagGraph(rootType reflect.Type, seen map[reflect.Type]*tagNode) *tagNod
 				} else {
 					tagValue = field.Name
 				}
-				if len(tagValue) != 0 {
-					// ignore any empty tags
+				if len(tagValue) != 0 && tagValue != "-" {
+					// ignore any empty tags and skipping fields
```

Comment:
Is this used in this PR? I didn't see any new structs with "-".

Reply:
This was used in the PR, but we no longer use it anymore.

I recalled we used to use `codec` in `SimulateRequest` to simplify the code (which turned out to be breaking the separation between endpoint and simulation logic), and `"-"` was in use to skip some fields from being unnecessarily considered in the tag graph test.

The tag graph test, if I understood correctly, was to ensure that the "structure" of pre-encodec-* is the same as generated model code's structure. The motivation behind is that, since generated `model.SimulateRequestTransactionGroup` is only `[]json.RawMessage`, we want to gain more info of txn group, thus introduce preencoded-* structs that gives more information about txn and handles (de)serialization.

Back to the tag graph test: we view both structs as "tree-alike" structure, and we want to check if the trees are the same up by tags attached to each node/leaf. We want to omit some cases, i.e., the field is omitted by tag `"-"`, and that is what this skip condition is for.

It is mostly test enhancement, but there's a trial of history and thinking behind...

Reply:
You're correct -- I don't think this is strictly necessary anymore, but this PR used to directly encode a simulation package struct in the REST response which had some hidden fields, which this test couldn't handle.

Still, I'd like to keep the improvement here, and @ahangsu also found and fixed a bug on line 121.